assets/chezmoi.io/docs/reference/templates/github-functions/gitHubLatestReleaseAssetURL.md

```diff
@@ -0,0 +1,20 @@
+# `gitHubLatestReleaseAssetURL` *owner-repo* *pattern*
+
+`gitHubLatestReleaseAssetURL` calls the GitHub API to retrieve the latest
+release about the given *owner-repo*, returning structured data as defined by
+the [GitHub Go API
+bindings](https://pkg.go.dev/github.com/google/go-github/v61/github#RepositoryRelease).
+It then iterates through all the release's assets, returning the first one that
+matches *pattern*. *pattern* is a shell pattern as [described in
+`path.Match`](https://pkg.go.dev/path#Match).
+
+Calls to `gitHubLatestReleaseAssetURL` are cached so calling
+`gitHubLatestReleaseAssetURL` with the same *owner-repo* will only result in one
+call to the GitHub API.
```

Comment:
The examples use `.chezmoi.os` and `.chezmoi.arch`, but not all releases use the go values presented by that. I wonder if it might be worth using a third example for a tool like `ripgrep`, written in Rust and therefore using the  triples (https://github.com/BurntSushi/ripgrep/releases/tag/14.1.0, e.g., aarch64-apple-darwin).

Reply:
Good point. Also, Debian and RedHat use different architecture and OS values to `GOARCH` and `GOOS`. I wonder if it's worth adding a few extra standard template variables like `.chezmoi.rpmArch`, `.chezmoi.debArch`, `.chezmoi.rustOS`, and `.chezmoi.rustArch`.

Reply:
If you’re going to do `rustOS`, you probably want to just do `rustArchTriple` since that’s the typical approach.